sdk/core/azure-core-http-netty/src/test/java/com/azure/core/http/netty/NettyHttpHeadersTests.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty;
+
+import com.azure.core.http.HttpHeaders;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class NettyHttpHeadersTests {
+    @Test
+    public void testSet() {
+        io.netty.handler.codec.http.HttpHeaders nettyHeaders = new DefaultHttpHeaders();
+
+        final HttpHeaders headers = new ReactorNettyHttpResponseBase.NettyHttpHeaders(nettyHeaders);
+
+        headers.set("a", "b");
+        assertEquals("b", headers.getValue("a"));
+
+        headers.set("a", "c");
+        assertEquals("c", headers.getValue("a"));
+
+        headers.set("a", (String) null);
+        assertNull(headers.getValue("a"));
+
+        headers.set("A", "");
+        assertEquals("", headers.getValue("a"));
```

Comment:
Is this something netty's HttpHeaders class is doing to make the keys case insensitive? I didn't see any `toLowerCase` calls in NettyHttpHeaders `get()` method.

Reply:
Right, we are case insensitive in both implementations, but we are using Netty's built-in case insensitive functionality and therefore not doing extra work in addition to that.